### ISSUE TITLE
Fuzzy search doesn't work

### ISSUE THREAD

**Issue by kjbrum** [angeloashmore/react-use-flexsearch#7]
I would like to be able to return results as the user types using fuzzy search. Is this possible?

Right now, when I type `dev` I don't get any results, but if I type `development` I get the results I expect. Maybe I'm just misunderstanding how FlexSearch works?

Thanks!

**Comment by angeloashmore** [comment]
Hi @kjbrum, I'm definitely not an expert with FlexSearch, but I think you should be able to adjust how fuzzy the search is using the search options.

Official docs: https://github.com/nextapps-de/flexsearch#custom-search

You can pass in custom search options through the fourth parameter of `useFlexSearch`: https://github.com/angeloashmore/react-use-flexsearch#api

**Comment by kjbrum** [comment]
Sounds like it should help, thanks!

**Comment by VedantLohbare** [comment]
Hi @kjbrum, 
Just wondering if you were able to achieve this and if so, do you mind helping me out?
I am trying to do this but I still cannot search for partial strings
```  
const searchOptions = { tokenize: "full" }
  const results = useFlexSearch(query, index, store, searchOptions)
```

**Comment by NickBarreto** [comment]
I'm having exactly the same issue @VedantLohbare. If you ever figured it out let me know. 

From poking around a bit in the code, it looks to me like the options in the fourth argument of `useFlexSearch` get passed to an `index.search`, but Search options don't include 'tokenize': https://github.com/nextapps-de/flexsearch#search-options

That tokenize: 'forward' or 'full' or whatever option would need to be passed in when the index is created, so maybe the option is going to be something that needs to be turned on in the flexsearch setup, which for me is happening with gatsby-plugin-local-search.

**Comment by DHFW** [comment]
@NickBarreto, @VedantLohbare thank you for your ideas. I was able to make it work (partial text search). In your `gatsby-plugin-local-search` set the `engineOptions` to:
```
engineOptions: {
  tokenize: 'full'
}
```

And in the useFlexSearch add the searchOptions: `{tokenize: 'full'}` as well. 

More details [here](https://archived.vercel.app/blog/adding-search)

**Comment by NickBarreto** [comment]
Hi @DHFW,

I came to the same conclusion. In fact I made a PR to update the docs in `gatsby-plugin-local-search` to make this a little more explicit (and because some of the options changed). [This part of the readme](https://github.com/angeloashmore/gatsby-plugin-local-search#flexsearch-options) goes through a few options that are analogous to the FlexSearch documentation's options. I hope it's helpful.
